examples/nlp/question_answering/question_answering_squad.py

```diff
@@ -26,19 +93,48 @@
 def main(cfg: DictConfig) -> None:
     logging.info(f'Config: {cfg.pretty()}')
     trainer = pl.Trainer(**cfg.trainer)
-    exp_manager(trainer, cfg.get("exp_manager", None))
+    exp_dir = exp_manager(trainer, cfg.get("exp_manager", None))
 
-    question_answering_model = QAModel(cfg.model, trainer=trainer)
-    trainer.fit(question_answering_model)
+    do_training = True
+    if not cfg.pretrained_model:
+        logging.info(f'Config: {OmegaConf.to_yaml(cfg)}')
+        model = QAModel(cfg.model, trainer=trainer)
+    else:
+        logging.info(f'Loading pretrained model {cfg.pretrained_model}')
+        # TODO: Remove strict, when lightning has persistent parameter support for add_state()
+        model = QAModel.from_pretrained(cfg.pretrained_model, strict=False)
+        if do_training:
+            model.setup_training_data(train_data_config=cfg.model.train_ds)
+            model.setup_validation_data(val_data_config=cfg.model.validation_ds)
+
+    if do_training:
+        trainer.fit(model)
+        if cfg.model.nemo_path:
+            model.save_to(cfg.model.nemo_path)
 
     if hasattr(cfg.model, 'test_ds') and cfg.model.test_ds.file is not None:
         gpu = 1 if cfg.trainer.gpus != 0 else 0
         trainer = pl.Trainer(gpus=gpu)
-        if question_answering_model.prepare_test(trainer):
-            trainer.test(question_answering_model)
+        model.setup_test_data(test_data_config=cfg.model.test_ds)
+        if model.prepare_test(trainer):
```

Comment:
When would prepare_test return False?

Reply:
https://github.com/NVIDIA/NeMo/blob/main/nemo/core/classes/modelPT.py#L877
This is good practice recommended from core team